resources/replace_or_add.rb

```diff
@@ -20,15 +21,15 @@
   # replace
   current.each do |line|
     line = line.dup
-    if line =~ regex || line == new_resource.line
+    if line =~ regex || line == add_line
       found = true
-      line = new_resource.line unless line == new_resource.line
+      line = add_line unless line == add_line
```

Comment:
I'm confused why the conditional in this line was needed in the first place and is needed now. What's wrong with a mere:

```rb
line = add_line
```